packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -110,12 +130,21 @@ class TransferEngine implements ITransferEngine {
         );
       }
 
+      // Update the destination provider's source metadata
+      const { source: sourceMetadata } = this.#metadata;
+
+      if (sourceMetadata) {
+        this.destinationProvider.setMetadata?.('source', sourceMetadata);
+      }
```

Comment:
It feels like this whole section should either be in bootstrap or (if that is reserved for only bootstrapping the providers) an initialization method

Reply:
Indeed, that makes sense. I've added an init step that is run after the providers' bootstrap